### PR TITLE
BigQuery: Support `AT TIME ZONE` in `EXTRACT`

### LINKED ISSUE #3003: BigQuery: `EXTRACT( ... AT TIME ZONE)` is unparsable
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Trying to run SQLFluff on an SQL that uses BigQuery `EXTRACT` Timestamp function using the optionla `AT TIME ZONE` variant returns unparsable errors.

### Expected Behaviour

SQLFluff should be able to parse and understand this syntax.

### Observed Behaviour

```
Line 1, Position 40: Found unparsable section: ' AT TIME ZONE some_timezone'
```

### How to reproduce

```
SELECT EXTRACT(HOUR FROM some_timestamp AT TIME ZONE some_timezone)
```

https://online.sqlfluff.com/fluffed?sql=H4sIAHMXTGIC_wt29XF1DlFwjQgJcnQO0fDwDw1ScAvy91Uozs9NjS_JzE0tLknMLVBwDFEI8fR1VYjy93NFyFXl56Vq8nIBAOxpD9hFAAAA&dialect=bigquery

### Dialect

BigQuery

### Version

0.11.2 -> Webversion 

### Configuration

Default

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by tunetheweb**
### Brief summary of the change made

Fixes #3003 

### Are there any other side effects of this change that we should be aware of?
I moved the `AtTimeZone` Segment from Postgres to ANSI as seems like quite a few dialects support this.

I also changed it form a Quoted Literal to an Expression as per comment below. This changed some of the Postgres YAMLs.

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_bigquery.py [hunk 1]
```diff
@@ -718,6 +718,19 @@ class SemiStructuredAccessorSegment(BaseSegment):
...
+class TimeZoneGrammar(BaseSegment):
+    """Casting to Time Zone.
+
+    BigQuery uses Double Quotes
+    https://cloud.google.com/bigquery/docs/reference/standard-sql/timestamp_functions#timezone_definitions
+    """
+
+    type = "time_zone_grammar"
+    match_grammar = AnyNumberOf(
+        Sequence("AT", "TIME", "ZONE", Ref("DoubleQuotedLiteralSegment")),
```
Comment (dmohns) [inline src/sqlfluff/dialects/dialect_bigquery.py:730]: In my use-case the timezone will come from a column in the table itself. I think this case is not covered here, right?

Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_bigquery.py:758]: I didn't know that was possible and thought your example was wrong. Will add that support.

Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_bigquery.py]: Added in https://github.com/sqlfluff/sqlfluff/pull/3004/commits/b5555a38c190dcd07ac056f8d9ac53294ed1b91a

File: src/sqlfluff/dialects/dialect_bigquery.py [hunk 2]
```diff
@@ -427,19 +444,30 @@ class FunctionSegment(ansi.FunctionSegment):
...
+                exclude=Ref("ExtractFunctionNameSegment"),
+            ),
+            Bracketed(
+                Delimited(
+                    Ref("DatetimeUnitSegment"),
+                    Ref(
+                        "FunctionContentsGrammar",
+                        ephemeral_name="FunctionContentsGrammar",
                     ),
                 ),
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_bigquery.py:472]: I de-dented this, as there was a pointless extra `Sequence` in there.

File: src/sqlfluff/dialects/dialect_ansi.py [hunk 3]
```diff
@@ -611,6 +611,15 @@ class ArrayLiteralSegment(BaseSegment):
     )
 
 
+class TimeZoneGrammar(BaseSegment):
+    """Casting to Time Zone."""
+
+    type = "time_zone_grammar"
+    match_grammar = AnyNumberOf(
+        Sequence("AT", "TIME", "ZONE", Ref("ExpressionSegment")),
```
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_ansi.py:619]: Seems like ColumnReferenceSegment and QuotedLiteralSegment are the only options here? Can we stick in a OneOf() for them, and kill the Expression?

Comment (WittierDinosaur) [reply src/sqlfluff/dialects/dialect_ansi.py:619]: Ah fair 👍

**Pull request merged**